query-node/mappings/src/bounty.ts

```diff
@@ -0,0 +1,485 @@
+      const isFunded = bounty.totalFunding >= fundingType.minFundingAmount
+      endFundingPeriod(store, bounty, isFunded)
+    }
+  })
+}
+
+export function bountyScheduleWorkSubmissionEnd(store: DatabaseManager, bounty: Bounty): void {
+  if (bounty.stage !== BountyStage.WorkSubmission) return
+
+  const workingPeriodEnd = fundingPeriodEnd(bounty) + bounty.workPeriod
+  scheduleAtBlock(workingPeriodEnd, () => {
+    if (bounty.stage === BountyStage.WorkSubmission) {
+      endWorkingPeriod(store, bounty)
+    }
+  })
+}
+
+export function bountyScheduleJudgementEnd(store: DatabaseManager, bounty: Bounty): void {
+  if (bounty.stage !== BountyStage.Judgment) return
+
+  const judgementPeriodEnd = fundingPeriodEnd(bounty) + bounty.workPeriod + bounty.judgingPeriod
+  scheduleAtBlock(judgementPeriodEnd, () => {
+    if (bounty.stage === BountyStage.Funding) {
+      bounty.updatedAt = new Date()
+      bounty.stage = BountyStage.Failed
+      store.save<Bounty>(bounty)
+    }
+  })
+}
+
+function endFundingPeriod(store: DatabaseManager, bounty: Bounty, isFunded = true): Promise<void> {
+  bounty.updatedAt = new Date()
+  if (isFunded) {
+    bounty.stage = BountyStage.WorkSubmission
+    bountyScheduleWorkSubmissionEnd(store, bounty)
+  } else {
+    bounty.stage = BountyStage[bounty.contributions?.length ? 'Failed' : 'Expired']
+  }
+  return store.save<Bounty>(bounty)
+}
+
+function endWorkingPeriod(store: DatabaseManager, bounty: Bounty): Promise<void> {
+  bounty.updatedAt = new Date()
+  if (bounty.entries?.length) {
+    bounty.stage = BountyStage.Judgment
+    bountyScheduleJudgementEnd(store, bounty)
+  } else {
+    bounty.stage = BountyStage.Failed
+  }
+  return store.save<Bounty>(bounty)
+}
+
+/**
+ * Event handlers
+ */
+
+// Store new bounties and their creation event
+export async function bounty_BountyCreated({ event, store }: EventContext & StoreContext): Promise<void> {
+  const createdEvent = new BountyEvents.BountyCreatedEvent(event)
+  const [bountyId, bountyParams, metadataBytes] = createdEvent.params
+  const eventTime = new Date(event.blockTimestamp)
+
+  const metadata = deserializeMetadata(BountyMetadata, metadataBytes)
+
+  // Create the bounty
+  const bounty = new Bounty({
+    id: String(bountyId),
+    createdAt: eventTime,
+    updatedAt: eventTime,
+    title: metadata?.title ?? undefined,
+    description: metadata?.description ?? undefined,
+    bannerImageUri: metadata?.bannerImageUri ?? undefined,
+    cherry: bountyParams.cherry,
+    entrantStake: bountyParams.entrant_stake,
+    creator: bountyActorToMembership(bountyParams.creator),
+    oracle: bountyActorToMembership(bountyParams.oracle),
+    fundingType: asFundingType(bountyParams.funding_type),
+    contractType: asContractType(bountyParams.contract_type),
+    workPeriod: bountyParams.work_period.toNumber(),
+    judgingPeriod: bountyParams.judging_period.toNumber(),
+
+    stage: BountyStage.Funding,
+    totalFunding: bountyParams.cherry,
```

Comment:
Right the cherry shouldn't be included in `totalFunding` since it's not included in the funding targets. Also I realized that it makes more sense for Pioneer to not have the funds subtracted on funding withdrawn or cash out.